sensorprocess/sensorprocess.go

```diff
@@ -0,0 +1,125 @@
+// Package sensorprocess contains the logic to add lidar or replay sensor readings to cartographer's mapbuilder
+package sensorprocess
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"math"
+	"time"
+
+	"github.com/edaniels/golog"
+	"go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/utils/contextutils"
+
+	"github.com/viamrobotics/viam-cartographer/cartofacade"
+	"github.com/viamrobotics/viam-cartographer/sensors/lidar"
+)
+
+// Config holds config needed throughout the process of adding a sensor reading to the mapbuilder.
+type Config struct {
+	Cartofacade      cartofacade.Interface
+	Lidar            lidar.Lidar
+	LidarName        string
+	DataRateMs       int
+	Timeout          time.Duration
+	Logger           golog.Logger
+	TelemetryEnabled bool
+	// addSensorReadingFromReplaySensor func(context.Context, []byte, time.Time, Config)
+	// addSensorReadingFromLiveReadings func(context.Context, []byte, time.Time, Config) int
+}
+
+// Start polls the lidar to get the next sensor reading and adds it to the mapBuilder.
+// stops when the context is Done.
+func Start(
+	ctx context.Context,
+	config Config,
+) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			addSensorReading(ctx, config)
+		}
+	}
+}
+
+// addSensorReading adds a lidar reading to the mapbuilder.
+func addSensorReading(
+	parentCtx context.Context,
+	config Config,
+) {
+	ctxWithMetadata, md := contextutils.ContextWithMetadata(parentCtx)
+	readingPc, err := config.Lidar.GetData(ctxWithMetadata)
+	if err != nil {
+		config.Logger.Warnw("Skipping sensor reading due to error getting lidar reading", "error", err)
```

Comment:
[note to self] this preserves the old behavior on getting an error reading from a lidar: https://github.com/viamrobotics/viam-cartographer/blob/f678204f865e7a25155a1a55972b92e0524ad1ea/viam-cartographer.go#L339